crates/typst/src/foundations/content.rs

```diff
@@ -961,18 +965,39 @@ pub trait PlainText {
     fn plain_text(&self, text: &mut EcoString);
 }
 
-/// The missing field access error message.
-#[cold]
-fn missing_field(field: &str) -> EcoString {
-    eco_format!("content does not contain field {}", field.repr())
+/// An error arising when trying to access a field of content.
+#[derive(Copy, Clone, Debug)]
+pub enum FieldAccessError {
+    MissingField,
+    MissingFieldNoDefault,
+    FieldNotSet,
+    Internal,
 }
 
-/// The missing field access error message when no default value was given.
-#[cold]
-fn missing_field_no_default(field: &str) -> EcoString {
-    eco_format!(
-        "content does not contain field {} and \
-         no default value was specified",
-        field.repr()
-    )
+impl FieldAccessError {
+    /// Gets the error message given a field name.
+    #[cold]
+    pub fn get_message(self, field: &str) -> EcoString {
+        match self {
+            FieldAccessError::MissingField => {
+                eco_format!("content does not contain field {}", field.repr())
+            }
+            FieldAccessError::MissingFieldNoDefault => {
+                eco_format!(
+                    "content does not contain field {} and \
+                 no default value was specified",
+                    field.repr()
+                )
+            }
+            FieldAccessError::FieldNotSet => {
+                eco_format!("field {} in content is not set at this point", field.repr())
```

Comment:
the wording here could be slightly confusing if a person has written a set rule right above and thinks that this will have an effect. perhaps "is not known" at this point? we could also add a hint, but I have no idea how to explain where it is available in the length of a hint.

Reply:
We might also want to reword the variants and message perhaps:

- `MissingField: content does not contain field {}` could be `Unknown: heading does not have field {}`

- `FieldNotSet: field {} in content is not set at this point` could be `Unset: field {} in heading is not yet known at this point`